### PR TITLE
operator: Making all pods and containers compliant with restricted policy

### PR DESCRIPTION
**Pull request by Red-GV**
**What this PR does / why we need it**:
With the release of the [Pod Security Admission](https://kubernetes.io/docs/concepts/security/pod-security-admission/) in 1.24/1.25 Kubernetes, this PR makes all pods and containers created by the operator (and the operator itself) compliant with the [restricted policy](https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted). These changes allow for compliance in native Kubernetes and OCP.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated
- [x] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/internal/manifests/build_test.go [hunk 1]
```diff
@@ -539,6 +539,139 @@ func TestBuildAll_WithFeatureFlags_EnableTLSGRPCServices(t *testing.T) {
...
+						},
+					},
+				},
+				Flags: FeatureFlags{
+					EnableRuntimeSeccompProfile: false,
+				},
+			},
+		},
+		{
+			desc: "disabled default/runtime seccomp profile",
```
Comment (periklis) [inline operator/internal/manifests/build_test.go:592]: typo `s/disabled/enabled`

**Review by periklis — COMMENTED** [review]
Looks good so far for me, besides testing on non-openshift environments, e.g. using `make quickstart` on kind

**Review by periklis — COMMENTED** [review]
Awesome! :rocket: Any status of your smoke tests around this?

**Review by periklis — COMMENTED** [review]
Good catch with the broker.

**Comment by Red-GV** [comment]
@periklis Smoke testing complete for Kind (v0.11), OCP 4.10, and OCP 4.11 pre-release. The `logcli` addon is working as expected. The `promtail` addon seems to not work correctly with an `emptyDir`. However, I'd like to address this as a followup PR if possible.

Just as a small summary after the rebase. Running on native K8s, the `runAsUser` field had to be set to a nonroot user (made it the user id from the operator's `Dockerfile`) since K8s could not determine the user's ID from the Loki image. On OCP, the `runAsUser` field is enforced through the SCC. Therefore, no additional actions need to be taken on that platform.

**Review by periklis — APPROVED** [review]

**Pull request merged**